File: libs/sdk/android/src/main/java/com/rudderstack/react/android/RudderSdkReactNativePackage.java
//package com.rudderstack.react.android;
//
//import com.facebook.react.BaseReactPackage;
//import com.facebook.react.bridge.NativeModule;
//import com.facebook.react.bridge.ReactApplicationContext;
//import com.facebook.react.module.model.ReactModuleInfo;
//import com.facebook.react.module.model.ReactModuleInfoProvider;
//import java.util.HashMap;
//import java.util.Map;
//
//public class RudderSdkReactNativePackage extends BaseReactPackage {
//  @Override
//  public NativeModule getModule(String name, ReactApplicationContext reactContext) {
//    System.out.println("Abhishek: Java");
//    if (name.equals(RudderSdkReactNativeModule.NAME)) {
//      return new RudderSdkReactNativeModule(reactContext);
//    } else {
//      return null;
//    }
//  }
//
//  @Override
//  public ReactModuleInfoProvider getReactModuleInfoProvider() {
//    return new ReactModuleInfoProvider() {
//      @Override
//      public Map<String, ReactModuleInfo> getReactModuleInfos() {
//        Map<String, ReactModuleInfo> moduleInfos = new HashMap<>();
//        moduleInfos.put(RudderSdkReactNativeModule.NAME, new ReactModuleInfo(
//                RudderSdkReactNativeModule.NAME,
//                RudderSdkReactNativeModule.NAME,
//                false,  // canOverrideExistingModule
//                false,  // needsEagerInit
//                false,  // isCxxModule
//                true    // isTurboModule
//        ));
//        return moduleInfos;
//      }
//    };
//  }
//}

File: libs/sdk/android/src/main/java/com/rudderstack/react/android/RudderSdkReactNativePackage.kt
package com.rudderstack.react.android;

import com.facebook.react.BaseReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.module.model.ReactModuleInfo
import com.facebook.react.module.model.ReactModuleInfoProvider
import java.util.HashMap

class RudderSdkReactNativePackage : BaseReactPackage() {
  override fun getModule(name: String, reactContext: ReactApplicationContext): NativeModule? {
    return if (name == RudderSdkReactNativeModule.NAME) {
      RudderSdkReactNativeModule(reactContext)
    } else {
      null
    }
  }

  override fun getReactModuleInfoProvider(): ReactModuleInfoProvider {
    return ReactModuleInfoProvider {
      val moduleInfos: MutableMap<String, ReactModuleInfo> = HashMap()
      moduleInfos[RudderSdkReactNativeModule.NAME] = ReactModuleInfo(
        RudderSdkReactNativeModule.NAME,
        RudderSdkReactNativeModule.NAME,
        false,  // canOverrideExistingModule
        false,  // needsEagerInit
        false,  // isCxxModule
        true // isTurboModule
      )
      moduleInfos
    }
  }
}
